### PR TITLE
develop_rating

### PR DESCRIPTION
**Pull request by tomjur2107**

### CODE REVIEW

File: PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.scss [hunk 1]
```diff
@@ -126,3 +128,95 @@ img {
...
+  flex-wrap: wrap;
+  width: 100%;
+  margin: 0em auto;
+  justify-content: center;
+}
+
+.ratebtn{
+  display: flex;
+  flex-wrap: wrap;
+  margin-left: 41.2%;
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.scss]: what are you trying to achieve here?

File: PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.scss [hunk 2]
```diff
@@ -126,3 +128,95 @@ img {
...
+.rate:not(:checked) > label:hover ~ label {
+  color: #deb217;  
+}
+.rate > input:checked + label:hover,
+.rate > input:checked + label:hover ~ label,
+.rate > input:checked ~ label:hover,
+.rate > input:checked ~ label:hover ~ label,
+.rate > label:hover ~ input:checked ~ label {
+  color: #c59b08;
+}
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.scss]: https://stackoverflow.com/questions/44960998/css-rating-css-system

File: PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts [hunk 3]
```diff
@@ -132,4 +166,55 @@ export class ProductComponent implements OnInit {
...
+    this.rating.rate = this.stars;
+    this.rating.productCode = this.param
+    this.rating.content = this.content.value
+    this.rating.userLogin = this.loginService.getLoggedUser().login
+    this.rating.userCode = this.loginService.getLoggedUser().code
+    this.rating.date = this.datePipe.transform(this.myDate, 'yyyy-MM-dd');
+    this.ratingSerivce.post(this.rating).subscribe(data => {
+      const modalRef = this.modalService.open(SuccessModalComponent);
+      modalRef.componentInstance.title = "Success!";
+      modalRef.componentInstance.message = "You rated product.";
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts]: I think modal here is unnecessary

File: PhotoPlusFrontend/PhotoPlus/src/app/services/rating/rating.service.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+export class RatingService extends AbstractService<Rating>{
+
+  constructor(http: HttpClient) {
+    super(http, "rating");
+  }
+
+  getRatingsPage(page: number, sortedBy: string, productCode: string) {
+    let params = new HttpParams().set("sortedBy", sortedBy)
+    params.set("productCode", productCode)
+    console.log(params)
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/services/rating/rating.service.ts]: remove console logs

File: PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.scss [hunk 5]
```diff
@@ -126,3 +128,95 @@ img {
...
 }
+
+.rate {
+  float: left;
+  height: 46px;
+  padding: 0 10px;
+}
+.rate:not(:checked) > input {
+  position:absolute;
+  top:-9999px;
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.scss]: why

**Review by dsuchy — CHANGES_REQUESTED** [review]

File: PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.scss [hunk 6]
```diff
@@ -136,7 +136,8 @@ img {
 }
 .rate:not(:checked) > input {
   position:absolute;
-  top:-9999px;
+  
+  left:-9999px;
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.scss]: W H Y

File: PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts [hunk 7]
```diff
@@ -200,6 +192,7 @@ export class ProductComponent implements OnInit {
       modalRef.componentInstance.message = "Please select stars!.";
       return;
     }
+    this.rating = new Rating;
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts]: how

Comment (dsuchy) [reply PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts]: how

**Review by dsuchy — CHANGES_REQUESTED** [review]

File: PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts [hunk 8]
```diff
@@ -193,10 +196,9 @@ export class ProductComponent implements OnInit {
       return;
     }
     this.rating = new Rating;
-    this.content = document.getElementById("area")
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts]: very nice keep it up

File: PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts (hunk 3 above)
Comment (syphzur) [reply PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts]: i think it is

Comment (syphzur) [reply PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts]: just leave it here

Comment (dsuchy) [reply PhotoPlusFrontend/PhotoPlus/src/app/components/product/product.component.ts]: ok

**Review by margawron — CHANGES_REQUESTED** [review]
Please merge with develop.

**Review by margawron — APPROVED** [review]

**Comment by margawron** [comment]
Improved styling, now it looks pretty good.

**Review by syphzur — APPROVED** [review]

**Review by dsuchy — APPROVED** [review]

**Pull request merged**